src/vivarium/framework/results/stratification.py

```diff
@@ -59,17 +62,29 @@ def __post_init__(self) -> None:
             raise ValueError("The sources argument must be non-empty.")
 
     def __call__(self, population: pd.DataFrame) -> pd.DataFrame:
+        """Apply the mapper to the population 'sources' columns and add the result
+        to the population. Any excluded categories (which have already been removed
+        from self.categories) will be converted to NaNs in the new column
+        and dropped later at the observation level.
+        """
         if self.is_vectorized:
-            raw_mapped_column = self.mapper(population[self.sources])
+            mapped_column = self.mapper(population[self.sources])
         else:
-            raw_mapped_column = population[self.sources].apply(self.mapper, axis=1)
-        mapped_column = raw_mapped_column.astype(
-            CategoricalDtype(categories=self.categories, ordered=True)
+            mapped_column = population[self.sources].apply(self.mapper, axis=1)
+
+        if mapped_column.isnull().any():
```

Comment:
Why is this check needed? Shouldn't it be sufficient to do the check below?

Reply:
I was trying to handle the fact that NaNs are not equal and so the set of NaNs would blow up the error message. I'll try and think of a way to handle that though b/c it would be nice to check against other unknown categories as well before raising.